doc/make_tables.py

```diff
@@ -0,0 +1,246 @@
+    header = """
+.. list-table:: Color schemes
+   :widths: 15 50 5 30
+   :header-rows: 1
+
+   * - Color scheme
+     - Description
+     - # colors
+     - Example
+"""
+    row_template = """
+   * - ``"{}"``
+     - {}
+     - {}
+     - .. image:: /{}
+"""
+
+    @classmethod
+    def fetch_tokens(cls):
+        return pv.charts._MultiCompPlot.COLOR_SCHEMES
+
+    @classmethod
+    def get_header(cls, tokens):
+        return cls.header
+
+    @classmethod
+    def get_row(cls, i, token, data):
+        if data["descr"] is None:
+            return None
+        else:
+            img_path = f"{CHARTS_IMAGE_DIR}/cs_{i}.png"
+            n_colors = cls.generate_img(token, img_path)
+            return cls.row_template.format(token, data["descr"], n_colors, img_path)
+
+    @staticmethod
+    def generate_img(color_scheme, img_path):
+        p = pv.Plotter(off_screen=True, window_size=[240, 120])
+        p.background_color = 'w'
+        chart = pv.Chart2D()
+        # Use a temporary plot to determine the total number of colors in this scheme
+        tmp_plot = chart.bar([0], [[1]] * 2, color=color_scheme, orientation="H")
+        n_colors = len(tmp_plot.colors)
+        plot = chart.bar([0], [[1]] * n_colors, color=color_scheme, orientation="H")
+        chart.remove_plot(tmp_plot)
+        plot.pen.color = 'w'
+        chart.x_range = [0, n_colors]
+        chart.hide_axes()
+        p.add_chart(chart)
+        _, img = p.show(screenshot=True, return_cpos=True)
+        # Crop the image and save it
+        p._save_image(img[34:78, 22:225, :], img_path, False)
+        return n_colors
+
+
+class ColorTable(DocTable):
+    """Class to generate colors table."""
+
+    path = f"{COLORS_TABLE_DIR}/colors.rst"
+    header = """
+.. list-table:: Colors
+   :widths: 50 20 30
+   :header-rows: 1
+
+   * - Name
+     - Hex value
+     - Example
+"""
+    row_template = """
+   * - {}
+     - ``{}``
+     - .. raw:: html
+
+          <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
+"""
+
+    @classmethod
+    def fetch_tokens(cls):
+        tokens = {token: {"hex": hex} for token, hex in pv.hexcolors.items()}
+        for s, token in pv.color_char_to_word.items():
+            if "synonyms" not in tokens[token]:
+                tokens[token]["synonyms"] = []
+            tokens[token]["synonyms"].append(s)
+        return tokens
+
+    @classmethod
+    def get_header(cls, tokens):
+        return cls.header
+
+    @classmethod
+    def get_row(cls, i, token, data):
+        token_template = '``"{}"``'
+        names = [token] + data.get("synonyms", [])
+        name = " or ".join(token_template.format(n) for n in names)
+        return cls.row_template.format(name, data["hex"], data["hex"])
+
+
+def make_all():
+    os.makedirs(CHARTS_IMAGE_DIR, exist_ok=True)
+    LineStyleTable.generate()
+    MarkerStyleTable.generate()
+    ColorSchemeTable.generate()
+    ColorTable.generate()
```

Comment:
No state indeed. In fact they used to be all separate methods before, but I wanted to group the different methods that construct each table 'somehow' and came up with this class structure. There might be better ways to do this.

Reply:
When I see stateless classes like this my first instinct is to just make them functions in a module, but the inheritance part actually adds to the functionality, so I think this is fine. But since we already have classes it's feasible to start using state for the paths. Again just a vague recommendation, current implementation works fine. If we don't expect anyone to use these classes outside this module then we don't have to complicate things.